### PR TITLE
Feat/euler diagram legend

### PR DESCRIPTION
**Pull request by jarvisraymond-uchicago**
Jira Ticket: [VADC-287](https://ctds-planx.atlassian.net/browse/VADC-287)

### New Features
This updates the Euler diagram to be more accessible,  and include both a text version and a legend. 

### Improvements
The text labels within the diagram's circle have been hidden, since they were overlapping when the cohort names were very long. 

This improves accessibility by using colors and borders with color contrast ratios sufficient to abide by the ADA in the diagram. Additionally a keyboard accessible text version is available now that should be able to be parsed by screen readers. 

Unit tests are included for the legend and text version. 

### Design 
image

### Implementation
<sub>Shown first using mouse to navigate, then with the keyboard</sub>
output

[VADC-287]: https://ctds-planx.atlassian.net/browse/VADC-287?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pieterlukasse — COMMENTED** [review]
thanks @jarvisraymond-uchicago ! Code looks good, as do the tests. I do have a comment about the name of one of the files. Please take a look.

Regarding the "switch" button
Screenshot 2023-02-23 at 17 35 00
I do have a few comments:
- Can we switch the sides of Diagram and Text Version? Given the Diagram comes first, I think it feels more natural if it is the left side option of the switch. 
- I think the blue/active color of the switch should be representing what is visible at that moment. Your implementation does the opposite, which I think might be a bit confusing/counter intuitive
   - This also means that the "disabled" side would need to be clickable.

**Comment by jarvisraymond-uchicago** [comment]
> 

Regarding: 
```Can we switch the sides of Diagram and Text Version? Given the Diagram comes first, I think it feels more natural if it is the left side option of the switch.```

Good idea, this is done now. 

Regarding: ```I think the blue/active color of the switch should be representing what is visible at that moment. Your implementation does the opposite, which I think might be a bit confusing/counter intuitive

    This also means that the "disabled" side would need to be clickable.
```
I don't agree with this - I don't think it is a good idea to have disabled looking buttons clickable in one instance and not another in the same application. However, I'm open to changing the UI to make it less confusing or counter intuitive. We could change the UI to have a switch (https://ant.design/components/switch/) with a label that changes based on its state or instead of having two buttons with different text, it could be updated to use only one button with text that changes. Or we could hold off on it.

File: src/Analysis/GWASV2/Components/Diagrams/CohortsOverlapDiagram/CohortsOverlapTextVersion.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    set13Size,
+    set23Size,
+    set123Size,
+    set1Label,
+    set2Label,
+    set3Label,
+  } = eulerArgs;
+  return (
+    <div className='euler-diagram-text-version'>
+      <h2>Cohort Intersections</h2>
```
[the remark this answers is not in the document]
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Components/Diagrams/CohortsOverlapDiagram/CohortsOverlapTextVersion.jsx]: Good idea, this is done now.

File: src/Analysis/GWASV2/Components/Diagrams/CohortsOverlapDiagram/CohortsOverlapTextVersion.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const CohortsOverlapTextVersion = ({ eulerArgs }) => {
```
[the remark this answers is not in the document]
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Components/Diagrams/CohortsOverlapDiagram/CohortsOverlapTextVersion.jsx]: Good idea, this is done now.

File: src/Analysis/GWASV2/Components/Diagrams/CohortsOverlapDiagram/Simple3SetsOverlapTextVersion.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const CohortsOverlapTextVersion = ({ title, eulerArgs }) => {
```
Comment (pieterlukasse) [inline src/Analysis/GWASV2/Components/Diagrams/CohortsOverlapDiagram/Simple3SetsOverlapTextVersion.jsx:4]: maybe rename this to match the file name?

Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASV2/Components/Diagrams/CohortsOverlapDiagram/Simple3SetsOverlapTextVersion.jsx]: Good catch, this has been updated.

**Review by pieterlukasse — APPROVED** [review]

**Pull request merged**